yaml config

sebasmannem opened the pull request:
some small fixes for SAN certs, and now working with yaml config.

sebasmannem committed:
SAN refinements

sebasmannem committed:
Fixes #2

Reading servers from ansible hosts file (yaml format)
Reading config from yaml file
helper script to bring it all together

cdorsman reviewed (changes requested):
Hi @sebasmannem,
Regarding __configFile variable: Would you consider the idea to have an separate variable configPath that will be overwritten be default ‘capath’ if it is None?

It gives a little bit more flexibility by default. Because you will be able to create more different configurations

cdorsman reviewed (changes requested):
Hi @sebasmannem Can you add some exception handling with the change on get_pem method?

Thnx!

sebasmannem commented:
> Hi @sebasmannem, Regarding __configFile variable: Would you consider the idea to have an separate variable configPath that will be overwritten be default ‘capath’ if it is None?
> 
> It gives a little bit more flexibility by default. Because you will be able to create more different configurations

We could, but we would also need to make that configurable through the yaml, and 
I don't think that having the configPath variable would be beneficial to the end goal (to have a TLS chain).
Do you have any specific reason for this to be configurable, or just because it could be?

sebasmannem commented:
By the way, I will also setup linting. I see a lot of things that should be improved if I just run flake8 and pylint.

cdorsman commented:
> By the way, I will also setup linting. I see a lot of things that should be improved if I just run flake8 and pylint.
Very good idea. In that same train of thought: Black as well?

cdorsman commented:
> > Hi @sebasmannem, Regarding __configFile variable: Would you consider the idea to have an separate variable configPath that will be overwritten be default ‘capath’ if it is None?
> > It gives a little bit more flexibility by default. Because you will be able to create more different configurations
> 
> We could, but we would also need to make that configurable through the yaml, and I don't think that having the configPath variable would be beneficial to the end goal (to have a TLS chain). Do you have any specific reason for this to be configurable, or just because it could be?

Yes. That the location of the configuration can be different than the expected default. And the idea that you would also serve multiple domains, with multiple configurations. It may get very messy.

sebasmannem committed:
pylint and some other fixes

sebasmannem committed:
A few things have changed:
- We now have a Config object that parses arguments, reads yaml config and returns values if they are in either, or ENV vars
- We now have a python application, installable with pip and unnable with `chainsmith` command
- We now have a docker container that can run chainsmith too

sebasmannem committed:
pylint and flake8 fixes

sebasmannem committed:
Enabling CI/CD python linting

sebasmannem committed:
publishing to PyPi

sebasmannem commented:
> > By the way, I will also setup linting. I see a lot of things that should be improved if I just run flake8 and pylint.
> 
> Very good idea. In that same train of thought: Black as well?

Yeah, there is an Action that can do all of them.
I ran it and ran into issues, so for now I manually run pylint and flake8.
In the future I will have another look at that to see where it brings us.
Or add Black to the pipeline I have right now.

sebasmannem commented:
> > > Hi @sebasmannem, Regarding __configFile variable: Would you consider the idea to have an separate variable configPath that will be overwritten be default ‘capath’ if it is None?
> > > It gives a little bit more flexibility by default. Because you will be able to create more different configurations
> > 
> > 
> > We could, but we would also need to make that configurable through the yaml, and I don't think that having the configPath variable would be beneficial to the end goal (to have a TLS chain). Do you have any specific reason for this to be configurable, or just because it could be?
> 
> Yes. That the location of the configuration can be different than the expected default. And the idea that you would also serve multiple domains, with multiple configurations. It may get very messy if the directory also contain other files than just the configurations.

We may need to talk about this, just to get a good understanding.

sebasmannem commented:
I now have chainsmith available on test.pypi.
SImilarly it will be added to pypi, but that happens automagically with a pipeline.
Let's merge this in, ok?

sebasmannem committed:
Adding hostname to SAN and registering IP as IP instead of DNS

sebasmannem committed:
Adding the ansible hosts option to arguments as --environment

sebasmannem committed:
- renamed environment to hosts
- few bugfixes
